### ISSUE TITLE
duplicated symbols in included external code

### ISSUE THREAD

**Issue by maltelenz** [ibpsa/modelica-ibpsa#1891]
The upcoming Modelica Specification comes with some clarifications regarding Include annotations:

https://specification.modelica.org/master/functions.html#S9.I4.i3.p1

https://github.com/modelica/ModelicaSpecification/issues/3451

In light of which these seem problematic:

https://github.com/ibpsa/modelica-ibpsa/blob/d78f71f340c81124590c561f58412c9624b5ad63/IBPSA/Utilities/IO/Files/BaseClasses/WeeklyScheduleObject.mo#L14

https://github.com/ibpsa/modelica-ibpsa/blob/d78f71f340c81124590c561f58412c9624b5ad63/IBPSA/Utilities/IO/Files/BaseClasses/WeeklyScheduleObject.mo#L34

https://github.com/ibpsa/modelica-ibpsa/blob/d78f71f340c81124590c561f58412c9624b5ad63/IBPSA/Utilities/IO/Files/WeeklySchedule.mo#L43

**Comment by mwetter** [comment]
@Mathadon : Can you please address this issue with the weekly schedule object.

**Comment by Mathadon** [comment]
@maltelenz do I understand correctly that this is the point of attention?

```
 The Include annotation shall be used in such a way that each external function can be handled in a separate translation unit. In particular, different external functions must not have Include annotations providing exported definitions of the same function symbol to avoid linking errors
```

I.e. we should create a separate header file for each function such that they can be compiled separately?

**Comment by maltelenz** [comment]
Yes, that is correct.

If we try to build these models with Wolfram System Modeler, and if we rely on the new specification, we get linking issues. Here is an example output on Linux:

```
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext1-destructor.o: In function `\``cmpfun':
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext1-destructor.c:(.text+0x0): multiple definition of `\``cmpfun'
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.o:WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.c:(.text+0x0): first defined here
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext1-destructor.o: In function `\``weeklyScheduleInit':
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext1-destructor.c:(.text+0x3c): multiple definition of `\``weeklyScheduleInit'
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.o:WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.c:(.text+0x3c): first defined here
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext1-destructor.o: In function `\``weeklyScheduleFreeInit':
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext1-destructor.c:(.text+0x130c): multiple definition of `\``weeklyScheduleFreeInit'
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.o:WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.c:(.text+0x130c): first defined here
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext1-destructor.o: In function `\``weeklyScheduleFree':
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext1-destructor.c:(.text+0x1397): multiple definition of `\``weeklyScheduleFree'
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.o:WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.c:(.text+0x1397): first defined here
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext1-destructor.o: In function `\``getScheduleValue':
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext1-destructor.c:(.text+0x1462): multiple definition of `\``getScheduleValue'
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.o:WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.c:(.text+0x1462): first defined here
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext2-getCalendarValue.o: In function `\``cmpfun':
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext2-getCalendarValue.c:(.text+0x0): multiple definition of `\``cmpfun'
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.o:WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.c:(.text+0x0): first defined here
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext2-getCalendarValue.o: In function `\``weeklyScheduleInit':
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext2-getCalendarValue.c:(.text+0x3c): multiple definition of `\``weeklyScheduleInit'
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.o:WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.c:(.text+0x3c): first defined here
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext2-getCalendarValue.o: In function `\``weeklyScheduleFreeInit':
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext2-getCalendarValue.c:(.text+0x130c): multiple definition of `\``weeklyScheduleFreeInit'
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.o:WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.c:(.text+0x130c): first defined here
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext2-getCalendarValue.o: In function `\``weeklyScheduleFree':
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext2-getCalendarValue.c:(.text+0x1397): multiple definition of `\``weeklyScheduleFree'
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.o:WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.c:(.text+0x1397): first defined here
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext2-getCalendarValue.o: In function `\``getScheduleValue':
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext2-getCalendarValue.c:(.text+0x1462): multiple definition of `\``getScheduleValue'
WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.o:WeeklySchedule7to19_2158263259_3163909_lyPCUHHRK_1717785048179_ext0-constructor.c:(.text+0x1462): first defined here
collect2: error: ld returned 1 exit status
```

**Comment by Mathadon** [comment]
@maltelenz does this branch work for you?
https://github.com/ibpsa/modelica-ibpsa/tree/1891_headers

**Comment by maltelenz** [comment]
@Mathadon I tried the branch in System Modeler.

We think the branch is now correct and following what the (upcoming) specification language says.

For System Modeler, we still cannot build, but this seems to be because we interpreted the `different external functions` formulation in the specification language a bit to narrowly, specifically with regards to multiple components of `IBPSA.Utilities.IO.Files.WeeklySchedule`, which contains the external function `getCalendarValue`.
